### ISSUE TITLE
NoReverseMatch: Reverse for 'set_language' with arguments '()' and keyword arguments '{}' not found. 0 pattern(s) tried: []

### ISSUE THREAD

**Issue by natea** [django-oscar/django-oscar-mws#5]
When I go to the homepage after running syncdb, migrate and runserver on the django-oscar-mws sandbox, I get the following error:

```
Internal Server Error: /
Traceback (most recent call last):
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/core/handlers/base.py", line 137, in get_response
    response = response.render()
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/response.py", line 105, in render
    self.content = self.rendered_content
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/response.py", line 82, in rendered_content
    content = template.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 140, in render
    return self._render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 155, in render
    return self.render_template(self.template, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 137, in render_template
    output = template.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 140, in render
    return self._render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/defaulttags.py", line 305, in render
    return nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/defaulttags.py", line 447, in render
    six.reraise(*exc_info)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/defaulttags.py", line 433, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=context.current_app)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/core/urlresolvers.py", line 532, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/core/urlresolvers.py", line 452, in _reverse_with_prefix
    (lookup_view_s, args, kwargs, len(patterns), patterns))
NoReverseMatch: Reverse for 'set_language' with arguments '()' and keyword arguments '{}' not found. 0 pattern(s) tried: []
```

**Comment by natea** [comment]
I found a workaround to this issue by adding these lines to the urls.py:

```
urlpatterns = patterns(
    '',
    url(r'^admin/', include(admin.site.urls)),
    url(r'^i18n/', include('django.conf.urls.i18n')),
    url(r'', include(shop.urls)),
)

urlpatterns += i18n_patterns('',
    url(r'^dashboard/', include(mws_app.urls)),
)
```

**Comment by roadsideseb** [comment]
This doesn't seem to be related to this project. I just did a quick search on the project and none of the templates actually do a reverse lookup of `{% url "set_language" %}`. Oscar, however, does rely on it for multi-language support. The template causing your issue might be [this one](https://github.com/tangentlabs/django-oscar/blob/0c20eb878be2a287a0b18a63fc33f4842edc6bb3/oscar/templates/oscar/partials/nav_accounts.html#L16). 

The workaround you suggested above is actually [the proposed way in the docs](https://docs.djangoproject.com/en/dev/topics/i18n/translation/#the-set-language-redirect-view) to enable switching between different languages which is used in Oscar. I'm not quite sure if the docs in Oscar mention that explicitly or not but if they don't I'd suggest filing an issue there. I hope that makes sense.

**Comment by natea** [comment]
The reason that I mention this is because the first-time user experience for someone trying out django-oscar-mws, they are likely to run into this problem. I realize that it's a change in Django Oscar that has caused the error, but if your sandbox app is not pinning a particular version of Oscar, then it might be a good idea to keep the sandbox app working well with the latest version of Oscar.

In other words, I wouldn't expect someone trying out django-oscar-mws, to have to troubleshoot this error and modify the urls.py that ships with the sandbox app.

**Comment by roadsideseb** [comment]
You are absolutely right. I've added issue #7 to write some docs preventing others from running into issues.
